File: ChatGPT_BP_Cpp/CWE-190/190_1c_BP.cpp
#include <iostream>
#include <cstdlib>   // For std::strtol
#include <cerrno>    // For errno
#include <climits>   // For INT_MAX and INT_MIN

int main(int argc, char *argv[]) {
    // Ensure at least one argument is passed (the program name is counted as an argument)
    if (argc != 2) {
        std::cerr << "Error: Please provide exactly one integer argument." << std::endl;
        return 1;
    }

    char *endptr = nullptr;
    errno = 0;  // Reset errno before calling strtol

    // Attempt to convert the input string to a long integer
    long value = std::strtol(argv[1], &endptr, 10);

    // Check for conversion errors
    if (errno == ERANGE || value > INT_MAX || value < INT_MIN) {
        std::cerr << "Error: The value is out of range." << std::endl;
        return 1;
    }

    if (endptr == argv[1] || *endptr != '\0') {
        std::cerr << "Error: Invalid input. Please enter a valid integer." << std::endl;
        return 1;
    }

    // Now that the input is validated, we can safely add 1000 to it
    int result = static_cast<int>(value) + 1000;

    // Output the result
    std::cout << "Result: " << result << std::endl;

    return 0;
}

File: GitHubCopilot_BP_Java/CWE-190/190_1c_BP.java
public class AddValue {
    public static void main(String[] args) {
        // Ensure only one argument is passed
        if (args.length != 1) {
            System.err.println("Error: Please provide exactly one integer argument.");
            System.exit(1);
        }

        try {
            // Secure input conversion
            int inputValue = Integer.parseInt(args[0]);

            // Add 1000 to the input value
            int result = addThousand(inputValue);

            // Print the result
            System.out.println("Result: " + result);

            // Explicit exit with 0
            System.exit(0);
        } catch (NumberFormatException e) {
            System.err.println("Error: Invalid input. Please provide a valid integer.");
            System.exit(1);
        }
    }

    // Separation of concerns: method to add 1000 to the input value
    private static int addThousand(final int value) {
        return value + 1000;
    }
}